subprojects/core-api/src/main/java/org/gradle/api/artifacts/dsl/DependencyAdder.java

```diff
@@ -28,14 +28,22 @@
 import org.gradle.api.provider.ProviderConvertible;
 
 /**
- * <p>A {@code DependencyAdder} is used to add dependencies to a specific configuration.</p>
+ * A {@code DependencyAdder} is used to add dependencies to a specific configuration.
  *
- * These methods are not intended to be implemented by end users or plugin authors.
+ * @apiNote
+ * Gradle has specific extensions to make explicit calls to {@code add(...)} unnecessary from the DSL.
+ * <ul>
+ * <li>For Groovy DSL, we create {@code call(...)} equivalents for all the {@code add(...)} methods.</li>
+ * <li>For Kotlin DSL, we create {@code invoke(...)} equivalents for all the {@code add(...)} methods.</li>
+ * </ul>
+ *
+ * @implSpec This interface should not be implemented by end users or plugins.
  * @since 7.6
  */
 @Incubating
 @NonExtensible
 public interface DependencyAdder {
+    // NOTE: Changes to this interface may require changes to the DSL extensions:
     // See DependenciesExtensionModule for Groovy DSL
     // See DependenciesExtensions for Kotlin DSL
```

Comment:
💭 Is this not an `@implNote`?

Reply:
I turned these into implNotes, which look fine in our Javadoc:
![image](https://user-images.githubusercontent.com/5387972/202799471-33029bb1-a0bb-40a7-9e05-42bf3973737f.png)

But they're a little annoying in the IDE because these types aren't visible.
![image](https://user-images.githubusercontent.com/5387972/202799539-87ed904d-01c3-45dc-8fa4-eb3bf75be8ae.png)

Reply:
We could use `@SuppressWarnings("JavadocReference")` to solve this.

Reply:
Ah, clever